libcloud/common/google.py

```diff
@@ -480,6 +482,62 @@ def refresh_token(self, token_info):
             new_token["refresh_token"] = token_info["refresh_token"]
         return new_token
 
+    @property
+    def _redirect_uri_with_port(self):
+        return self.redirect_uri + ":" + str(self.redirect_uri_port)
+
+    def _receive_code_through_local_loopback(self):
```

Comment:
I added a test case, I had to move one of the mocks in GoogleTestCase out of the parent class and into the child classes that actually require it, because that mocked the behavior I actually want to test here.
The test is more or less end to end, I am not mocking the HTTP server, just the user interaction of signing in is replaced with a fake response from google